### ISSUE TITLE
ImageSwap 1.4.2 inherits vulnerabilities from image python:3-alpine thats was latest at build time

### ISSUE THREAD

**Issue by log2** [phenixblue/imageswap-webhook#60]
**What would you like to be added**:

A fixed tag (not a moving tag, such as `latest` or `python:3-alpine`) base image for Dockerfiles, in order for the base image to be easily inspectables for vulnerabilities.

Also, final imageswap and imageswap-init images should be free of vulnerabilities, at least those marked CRITICAL and HIGH.

A vulnerability-free image on `python:3-latest` timeline is `python:3.8.12-alpine3.15` (if the projects must stick to python `3.8`, specified in `python_version` of `requires` section in `Pipfile`s), or `python:3.11.0a3-alpine3.15` (if imageswap can bump to Python 3.11).

**Why is this needed**:

Current image (v.1.4.2) has various vulnerabilities that can be inspected via:

* `trivy image thewebroot/imageswap-init:v1.4.2`  [trivy-report-imageswap-init.txt](https://github.com/phenixblue/imageswap-webhook/files/7821460/trivy-report-imageswap-init.txt)
* `trivy image thewebroot/imageswap:v1.4.2` [trivy-report-imageswap.txt](https://github.com/phenixblue/imageswap-webhook/files/7821462/trivy-report-imageswap.txt)

Most of those vulnerabilities are inherited by the `ptyhon:3-alpine` image, being the latest when `v1.4.2` of imageswap was built.

Note that fixing vulnerabilities in base image might not be enough, since some vulnerabilities derive from python packages:
* in imageswap-init, kubernetes 11.0.0 [CVE-2020-1747] should be upgraded to 12.0.1 (at least)
* in imageswap, werkzeug 2.0.1 [CVE pyup.io-42050] could be upgraded to 2.0.2 (albeit this "vulnerability" affects just SameSite cookie, not important in this context)

However, those could be addressed in a different issue.

**Comment by phenixblue** [comment]
These CVE's should be resolved in the latest v1.5.0 release of ImageSwap

**Comment by log2** [comment]
For imageswap-init I can confirm no high or critical vulnerability are found with grype (switched to it from trivy, during these months):

```
❯ grype thewebroot/imageswap-init:v1.5.0 --add-cpes-if-none --only-fixed --fail-on high
 ✔ Vulnerability DB        [updated]
 ✔ Parsed image
 ✔ Cataloged packages      [72 packages]
 ✔ Scanned image           [6 vulnerabilities]
No vulnerabilities found
```

However, main image still suffers some high vulnerabilities:

```
❯ grype thewebroot/imageswap:v1.5.0 --add-cpes-if-none --only-fixed --fail-on high
 ✔ Vulnerability DB        [no update available]
 ✔ Parsed image
 ✔ Cataloged packages      [65 packages]
 ✔ Scanned image           [2 vulnerabilities]
NAME          INSTALLED  FIXED-IN   TYPE  VULNERABILITY  SEVERITY
libcrypto1.1  1.1.1o-r0  1.1.1q-r0  apk   CVE-2022-2097  High
libssl1.1     1.1.1o-r0  1.1.1q-r0  apk   CVE-2022-2097  High
1 error occurred:
	* discovered vulnerabilities at or above the severity threshold
```

Note that, for the latter image, trivy detects same vulnerabiities found by grype, plus another "disputed" and critical vulnerability in workzeug Python package:

```
❯ trivy image --severity HIGH,CRITICAL thewebroot/imageswap:v1.5.0
2022-07-18T09:35:43.259+0200	INFO	Need to update DB
2022-07-18T09:35:43.259+0200	INFO	DB Repository: ghcr.io/aquasecurity/trivy-db
2022-07-18T09:35:43.259+0200	INFO	Downloading DB...
33.07 MiB / 33.07 MiB [---------------------------------------------------------------------------------------------] 100.00% 17.06 MiB p/s 2.1s
2022-07-18T09:35:46.611+0200	INFO	Vulnerability scanning is enabled
2022-07-18T09:35:46.611+0200	INFO	Secret scanning is enabled
2022-07-18T09:35:46.611+0200	INFO	If your scanning is slow, please try '--security-checks vuln' to disable secret scanning
2022-07-18T09:35:46.611+0200	INFO	Please see also https://aquasecurity.github.io/trivy/v0.29.2/docs/secret/scanning/#recommendation for faster secret detection
2022-07-18T09:35:51.466+0200	INFO	Detected OS: alpine
2022-07-18T09:35:51.467+0200	INFO	Detecting Alpine vulnerabilities...
2022-07-18T09:35:51.469+0200	INFO	Number of language-specific files: 1
2022-07-18T09:35:51.469+0200	INFO	Detecting python-pkg vulnerabilities...

thewebroot/imageswap:v1.5.0 (alpine 3.16.0)

Total: 2 (HIGH: 2, CRITICAL: 0)

┌──────────────┬───────────────┬──────────┬───────────────────┬───────────────┬──────────────────────────────────────────────┐
│   Library    │ Vulnerability │ Severity │ Installed Version │ Fixed Version │                    Title                     │
├──────────────┼───────────────┼──────────┼───────────────────┼───────────────┼──────────────────────────────────────────────┤
│ libcrypto1.1 │ CVE-2022-2097 │ HIGH     │ 1.1.1o-r0         │ 1.1.1q-r0     │ openssl: AES OCB fails to encrypt some bytes │
│              │               │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-2097    │
├──────────────┤               │          │                   │               │                                              │
│ libssl1.1    │               │          │                   │               │                                              │
│              │               │          │                   │               │                                              │
└──────────────┴───────────────┴──────────┴───────────────────┴───────────────┴──────────────────────────────────────────────┘
2022-07-18T09:35:51.471+0200	INFO	Table result includes only package filenames. Use '--format json' option to get the full path to the package file.

Python (python-pkg)

Total: 1 (HIGH: 0, CRITICAL: 1)

┌─────────────────────┬────────────────┬──────────┬───────────────────┬───────────────┬─────────────────────────────────────────────────────────────┐
│       Library       │ Vulnerability  │ Severity │ Installed Version │ Fixed Version │                            Title                            │
├─────────────────────┼────────────────┼──────────┼───────────────────┼───────────────┼─────────────────────────────────────────────────────────────┤
│ Werkzeug (METADATA) │ CVE-2022-29361 │ CRITICAL │ 2.0.1             │ 2.1.1         │ ** DISPUTED ** Improper parsing of HTTP requests in Pallets │
│                     │                │          │                   │               │ Werkzeug v2.1.0...                                          │
│                     │                │          │                   │               │ https://avd.aquasec.com/nvd/cve-2022-29361                  │
└─────────────────────┴────────────────┴──────────┴───────────────────┴───────────────┴─────────────────────────────────────────────────────────────┘
```

Could you fix them too, please? Thank you

**Comment by phenixblue** [comment]
It looks like the libcrypto and libssl vulnerabilities are both related to the Alpine Linux base image. Alpine just released 3.16.1 a few days ago, so those fixes should get picked up with the next release. 

I don't see the Werkzeug vulnerability reported anymore, so it looks like that was disputed and removed.
